core/types/shard_state.go

```diff
@@ -7,28 +7,31 @@ import (
 	"golang.org/x/crypto/sha3"
 )
 
-// NodeID is a unique ID represent a node
-type NodeID string
+// NodeInfo represents a node info
+type NodeInfo struct {
+	NodeID   string
+	IsLeader bool
```

Comment:
If we define data structure like this:  shardID->[leaderID, list_of_nodes_in_this_shardID], it will be more efficient. But I am OK with the current design

Reply:
What is the point of adding a IsLeader boolean here? I don't see how it's affecting the logic? Basically the first node after random shuffle is automatically the leader. what's the reason we mark them out explicitly?

Reply:
good point

Reply:
i just modified the NodeInfo to NodeID which is a string. I still felt that we need to have indication to know who is the leader rather than using the first element of NodeList as the leader. it is because 
1/ if we do some testing for shuffle func - without any indication, we can not know who was the leader.
2/ hard to explain, but it's weird to consider the first element is the leader even I was the one who modified our resharding calcuation and designed the first element as the leader. with that, i just introduced one field to indicate the leader.

Reply:
ok, fair point